atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/KeyValueService.java

```diff
@@ -426,6 +426,17 @@ void deleteAllTimestamps(TableReference tableRef,
             TableReference tableRef,
             CandidateCellForSweepingRequest request);
 
+    /**
+     * Scans a range of a table and returns the row keys in that range.
+     * <p>
+     * This endpoint specifically avoids paging through each full row.
+     * <p>
+     * The batch size limits the number of results from each request.
+     */
+    default List<byte[]> getRowKeysInRange(TableReference tableRef, byte[] startRowInclusive, int batchSize) {
```

Comment:
From an API standpoint, I'd expect this to probably have the same interface as `getRange`, which takes a `RangeRequest` object (rather than always being end-unbounded). I see that the internal APIs e.g. for `GetCellTimestamps` are similarly always end-unbounded; these were in the past only used for Sweep, so end-unbounded made sense then, though if we're opening this up I don't know if it would be useful to allow end bounds. For the specific use case mentioned it's probably fine.

If I'm reading the impl correctly this actually returns the results of just one batch, so I think the `batchSize` should probably be something stronger, like `maxResults`

Reply:
@gmaretic - worth incorporating this feedback. I think taking an end byte[] is a good idea. Taking a `RangeRequest` unfortunately allows selecting columns to retain (ignored) and keeps the `batchHint` issue.

Reply:
I see you already swapped to `maxResults` - we can either also include the upper bound or can create a wrapper object to encapsulate them.